google/cloud/storage/internal/get_bucket_metadata_request.cc

```diff
@@ -0,0 +1,33 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/internal/get_bucket_metadata_request.h"
+#include <iostream>
+
+namespace google {
+namespace cloud {
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+namespace internal {
+std::ostream& operator<<(std::ostream& os, GetBucketMetadataRequest const& r) {
+  os << "GetBucketMetadataRequest={bucket_name=" << r.bucket_name();
```

Comment:
Do you want to put some additional string after the bucket name, so it doesn't run together with the subsequent field dump (which is what I assume DumpParameters does on proto bufs)?

Also, is DumpParameters different from DebugString on internally generated protobuf code (which would include the bucket name from the request, so you wouldn't need to add it separately).

Reply:
> Do you want to put some additional string after the bucket name, so it doesn't run together with the subsequent field dump

The `r.DumpParameters()` function will print a `, ` on the first non-empty parameter, 

> (which is what I assume DumpParameters does on proto bufs)?

There are no protobufs here, these are all plain C++ classes.

> Also, is DumpParameters different from DebugString on internally generated protobuf code (which would include the bucket name from the request, so you wouldn't need to add it separately).

Hmmm... Maybe I should rename these functions `DebugString()`?  They serve more or less the same purpose as `DebugString()` for protobufs.  What do you think?

Reply:
Given what you describe I wouldn't rename it DebugString, since the latter does a recursive proto message dump while your function just dumps top-level params.